string name
int32 state
---
uint32 ret